lib/presentation/screens/activity/views/map/activity_map_view.dart

```diff
@@ -0,0 +1,110 @@
+import 'package:movna/presentation/widgets/loading_indicator.dart';
+import 'package:movna/presentation/widgets/none_widget.dart';
+
+/// Displays a map with location information about the current activity.
+///
+/// Initializes the map at the last known location using [GetLastCoords] or to
+/// [GpsCoordinates.paris] if no previous location could be found.
+class ActivityMapView extends StatefulWidget {
+  const ActivityMapView({super.key});
+
+  @override
+  State<ActivityMapView> createState() => _ActivityMapViewState();
+}
+
+class _ActivityMapViewState extends State<ActivityMapView>
+    with TickerProviderStateMixin {
+  /// Controller use to modify and animate the [FlutterMap].
+  late final _controller = AnimatedMapController(
+    vsync: this,
+    duration: MapConstants.mapAnimationsDuration,
+    curve: MapConstants.mapAnimationsCurve,
+  );
+
+  @override
+  void initState() {
+    // Get last coords and set controller to the given coordinates or to Paris
+    injector<GetLastCoords>()().then(
+      (value) {
+        value.fold(
+          (lastCoords) {
+            _controller.move(
+              lastCoords.toLatLng(),
+              16,
+            );
+          },
+          (f) {
+            _controller.move(
+              GpsCoordinates.paris.toLatLng(),
+              6,
+            );
+          },
+        );
+      },
+    );
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocListener<ActivityCubit, ActivityState>(
+      listener: (context, state) {
+        // Listen to location changes and animate the map to the location
+        // (do not change zoom level)
+        final location =
+            state.mapOrNull(loaded: (loaded) => loaded.currentLocation);
+        if (location != null) {
+          _controller.animateTo(
+            dest: location.gpsCoordinates.toLatLng(),
+          );
+        }
+      },
+      child: FlutterMap(
+        mapController: _controller,
+        options: MapOptions(
+          center: GpsCoordinates.paris.toLatLng(),
+          zoom: 6,
+          maxZoom: MapConstants.maxZoom,
+        ),
+        children: [
+          TileLayer(
+            urlTemplate: MapConstants.urlTemplate,
+            // TODO constant or usecase https://github.com/MovnaTeam/movna/issues/21
+            userAgentPackageName: 'dev.movna.app',
+            tileProvider: injector<FMTCTileProvider>(),
+          ),
+          const UserLocationMarker(),
+          BlocBuilder<ActivityCubit, ActivityState>(
+            buildWhen: (prev, next) => prev.runtimeType != next.runtimeType,
+            builder: (context, state) {
+              return state.maybeMap(
+                loading: (_) => const Center(
+                  child: LoadingIndicator(),
+                ),
+                orElse: () => const NoneWidget(),
+              );
+            },
+          ),
```

Comment:
So if I understand correctly, you're hiding all the previous layers with the loading indicator when the state is loading, am I correct ? If so, wouldn't it be cleaner to inverse the logic and put the FlutterMap widget under a BlocBuilder that only build it when the state is loaded, and that puts a loading indicator when the state is loading ?

Reply:
the idea is to show the map whether we the location is loaded or not.
When the state does not have a location, the map is centered on the last known location.
If the state is loading the loading indicator is shown but it does not hide layers below, it is just shown on top of everythin